File: MyBackEnd/src/main/java/com/example/mybackend/Models/Inventory.java
package com.example.mybackend.Models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Inventory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String itemName;
    private int quantity;
    private double price;
    private double minQuantity;

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    @JsonIgnoreProperties("inventories")
    private InventoryCategory category;

    @ManyToMany
    @JoinTable(
            name = "inventory_supplier",
            joinColumns = @JoinColumn(name = "inventory_id"),
            inverseJoinColumns = @JoinColumn(name = "supplier_id")
    )
    @JsonIgnoreProperties("inventories")
    private List<Supplier> suppliers;

    @Override
    public String toString() {
        return "Inventory{" +
                "id=" + id +
                ", itemName='" + itemName + '\'' +
                ", quantity=" + quantity +
                ", price=" + price +
                ", category=" + (category != null ? category.getId() : "null") +  // Avoid circular reference
                ", minQuantity=" + minQuantity +
                ", suppliers=" + (suppliers != null ? suppliers.size() : "null") +  // Avoid circular reference
                '}';
    }
}

File: WebTs/src/types/Inventory.ts
import { Category } from "./Category";
import { Supplier } from "./Supplier";

export interface InventoryType {
  id: number;
  itemName: string;
  quantity: number;
  price: number;
  category: Category;
  minQuantity: number;
  suppliers: Supplier[];
}

export const emptyInventory: InventoryType = {
  id: 0,
  itemName: "",
  quantity: 0,
  price: 0,
  category: { id: 0, name: "" },
  minQuantity: 0,
  suppliers: [],
};
